src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [this.a, this.b, this.c].sort((first, second) => second - first);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('The sides do not form a valid triangle');
+    }
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Put number bigger than 0');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('Put radius bigger than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.pow(this.radius, 2) * Math.PI;
+
+    return Math.floor(area * 100) / 100;
```

Comment:
The `getArea` method in the `Circle` class uses `Math.floor` for rounding, which might not be consistent with the `Triangle` class that uses `Math.round`. Consider using the same rounding method for consistency.